minimal-plugin/src/main/java/io/micronaut/gradle/MicronautComponentPlugin.java

```diff
@@ -300,7 +300,10 @@ private static TaskProvider<ApplicationClasspathInspector> registerInspectRuntim
             var javaPluginExtension = PluginsHelper.javaPluginExtensionOf(project);
             task.setGroup(BasePlugin.BUILD_GROUP);
             task.setDescription("Performs sanity checks of the runtime classpath to warn about misconfigured builds");
-            task.getRuntimeClasspath().from(project.getConfigurations().getByName(JavaPlugin.RUNTIME_CLASSPATH_CONFIGURATION_NAME));
+            // Sort the runtimeClasspath before setting it since the order doesn't actually matter for inspection, however it does matter for caching.
+            task.getRuntimeClasspath().from(project.getConfigurations()
+                    .getByName(JavaPlugin.RUNTIME_CLASSPATH_CONFIGURATION_NAME).resolve()
```

Comment:
does this cause eager dependency configuration? that could be bad for performance.

Reply:
:x: This has indeed to be changed, configurations should never be resolved eagerly. For this check it doesn't really matter anyway to have sorted entries.

Reply:
That is correct, however order matters for calculating the task inputs fingerprint, which is used for caching. I'll fix this ofc.

Reply:
If the order is different then it means that the graph is different, so to me it makes sense that the task is out of date.